app/src/lib/editors/darwin.ts

```diff
@@ -0,0 +1,167 @@
+import * as Path from 'path'
+
+/**
+ * appPath will raise an error if it cannot find the program.
+ */
+const appPath: (bundleId: string) => Promise<string> = require('app-path')
+
+import {
+  LookupResult,
+  FoundEditor,
+  pathExists,
+  AtomLabel,
+  VisualStudioCodeLabel,
+  SublimeTextLabel,
+} from './shared'
+
+async function findAtomApplication(): Promise<LookupResult> {
+  const name = AtomLabel
+
+  try {
+    const installPath = await appPath('com.github.atom')
+    const path = Path.join(
+      installPath,
+      'Contents',
+      'Resources',
+      'app',
+      'atom.sh'
+    )
+    const exists = await pathExists(path)
```

Comment:
is this really finding if atom is installed?  or just that the shell script exists?